### PR TITLE
OEL-461: Add offcanvas pattern to search facets block.

### PR DESCRIPTION
**Pull request by GilNovacomm**

### CODE REVIEW

File: templates/patterns/offcanvas/pattern-offcanvas.html.twig [hunk 1]
```diff
@@ -0,0 +1,26 @@
+{#
```
Comment (drishu) [inline templates/patterns/offcanvas/pattern-offcanvas.html.twig]: remove this, we dont declare twice

File: templates/patterns/offcanvas/offcanvas.ui_patterns.yml [hunk 2]
```diff
@@ -0,0 +1,50 @@
+offcanvas:
```
Comment (drishu) [inline templates/patterns/offcanvas/offcanvas.ui_patterns.yml]: remove

File: templates/overrides/search/block--facets-form.html.twig [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+ *
+ * @see template_preprocess_block()
+ *
+ * @ingroup themeable
+ */
+#}
+
+{% block content %}
+  {{ pattern('offcanvas', {
+    'title': label,
```
Comment (drishu) [inline templates/overrides/search/block--facets-form.html.twig]: add extra classes and such, see comment in oe_bootstrap PR

**Review by drishu — CHANGES_REQUESTED** [review]

File: templates/overrides/search/block--facets-form.html.twig [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+      'heading': 4,
+      'extra_classes': "mb-lg-4",
+     },
+    'attributes': attributes.addClass('bcl-offcanvas'),
+    'offcanvas_id': 'bcl-offcanvas',
+    'body': content,
+    'placement': 'start',
+    'with_body_scroll': false,
+    'with_backdrop': true,
+    'extra_classes_body': 'bcl-offcanvas',
```
Comment (drishu) [inline templates/overrides/search/block--facets-form.html.twig]: add pl-g-0

**Review by drishu — CHANGES_REQUESTED** [review]

File: templates/overrides/search/block--facets-form.html.twig [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+{% block content %}
+  {{ pattern('offcanvas', {
+    'title': {
+      'label': label,
+      'heading': 4,
+      'extra_classes': "mb-lg-4",
+     },
+    'attributes': attributes.addClass('bcl-offcanvas'),
+    'offcanvas_id': 'bcl-offcanvas',
+    'body': content,
```
Comment (drishu) [inline templates/overrides/search/block--facets-form.html.twig:55]: check if we can override content.form.actions.reset and use the link (or button?) pattern, if not we need to preprocess the block, worst case we form alter and theme the reset from there

**Review by drishu — CHANGES_REQUESTED** [review]

**Review by drishu — APPROVED** [review]

**Review by timpevi — APPROVED** [review]

**Pull request merged**